### ISSUE TITLE
Output rarely produces incomplete line

### ISSUE THREAD

**Issue by kaskarn** [MicrosoftGenomics/FaST-LMM#12]
I have been running FaST on our UNIX cluster for a while, and recently got around to collecting results, using R and the data.table package.

The issue I am facing is that roughly one in twenty output files have an incomplete line, causing data.table::fread() to throw an error. To illustrate with chromosome 3 in our GWAS, the error is:

```
Error in checkForRemoteErrors(val) : 
  one node produced an error: Expected sep ('	') but new line or EOF ends field 8 on line 1332885 when reading data: 023	-	0.0856763	0.114	48.3307	0	0	3.36174e-06
```

Inspecting the file reveals that a valid line was incompletely written near the end (removing it manually allows fread() to read the file just fine).

I ran other analyses with different traits, and found the same error pop up, for different chromosomes and at different file location (though it seems always near the end).

I haven't looked at the FaST source code, but it might be worth checking if all write-caching is flushed consistently.

**Comment by CarlKCarlK** [comment]
Thanks for the report and thanks for using FaST-LMM.
Can you confirm that you're running the Python version? Also, can you tell us which function you're using (e.g. single_snp).
Thanks, Carl
p.s. Also, as an aside: would you be interested in a way to run "single_snp" with it spliting the work into as many pieces as you like (e.g. 0 of 200, 1 of 200, ... 199 of 200), with each piece writing a binary results file, and then a call to put the binary files together into the final text output file.

**Comment by kaskarn** [comment]
Hi Carl,

Thank you for taking the time to reply. Embarrassingly, I got the wrong FAST program -- the tool I am using was another GWAS tool called FAST (https://bitbucket.org/baderlab/fast/wiki/Home), and I am afraid I bothered you for nothing.

With that said, I enjoyed reading up on FaST-LMM, and I am tempted to try it alongside our typical approach. Regarding your comment: In my humble opinion, features further parallelizing single-SNP  GWAS would be welcome and, in general, are worth implementing wherever possible.

Cheers,
Antoine
